sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -126,14 +127,56 @@ class SessionCatalog(
   }
 
   /**
-   * A cache of qualified table names to table relation plans.
-   */
+    * A cache of qualified table names to table relation plans.
+    * Accessing tableRelationCache directly is not recommended,
+    * since it will introduce exposures to guava libraries.
+    */
   val tableRelationCache: Cache[QualifiedTableName, LogicalPlan] = {
     val cacheSize = conf.tableRelationCacheSize
     CacheBuilder.newBuilder().maximumSize(cacheSize).build[QualifiedTableName, LogicalPlan]()
   }
 
   /**
+    * This method provides a way to get a cached plan
```

Comment:
The "without exposing components to Guava" part of all these comments is unnecessary.